native/e2e/record.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const EventEmitter = require('events');
+const { cleanFileName } = require('./utils');
+
+const AndroidRecorder = require('./recorders/android-recorder');
+const IOSRecorder = require('./recorders/ios-recorder');
+
+const Recorder =
+  process.env.BPK_PLATFORM === 'android' ? AndroidRecorder : IOSRecorder;
+
+module.exports = storybookController => async (
+  kind,
+  story,
+  basePath,
+  testFuction,
+) => {
+  const videoName = cleanFileName(story, kind);
+  const recorder = new Recorder(`${basePath}/${videoName}`, videoName);
+  const testEmitter = new EventEmitter();
+
+  testEmitter.on('test:started', () => recorder.startRecording());
+  testEmitter.on('test:ended', async () => {});
+
+  await storybookController.start();
+  await storybookController.activateStory(kind, story);
+
+  await testFuction(testEmitter);
+
+  await recorder.finishRecording();
+  await storybookController.done();
+};
```

Comment:
Quick note, the two functions above cannot be excuted as callback to an event, the origin implementation was

```
estEmitter.on('test:ended', async () => {
    await recorder.finishRecording();
    await storybookController.done();
});
```

but the process exits before that is completed and the result is videos are not saving and the test command hangs forever, so I've moved it and (a)waiting for it to complete

I'll remove the test ended in the next commit, for now it's just a reference